### PR TITLE
Implemented Transpose for Ragged<T>

### PR DESCRIPTION
**Pull request by qindazhu**
And moved `Stack<T>` to `ragged.h`

### CODE REVIEW

File: k2/csrc/ragged.h [hunk 1]
```diff
@@ -338,6 +338,53 @@ struct Ragged {
 template <typename T>
 std::ostream &operator<<(std::ostream &stream, const Ragged<T> &r);
 
+/*
+  Stack a list of Ragged arrays to create a Ragged array with one more axis.
```
Comment (danpovey) [inline k2/csrc/ragged.h:342]: Thanks, can you please remove the `const` from Ragged and RaggedShape inputs?  I want to get out of the habit of using that.  Getting row-ids/row-splits is not always const.

File: k2/csrc/ragged.h [hunk 2]
```diff
@@ -338,6 +338,53 @@ struct Ragged {
...
+  Similar to TF/PyTorch's Stack.  The result will have Dim0 == num_srcs.  All
+  the source Ragged arrays' shapes must have the same NumAxes().
+
+     @param [in] axis   The new axis whose dimension will equal num_srcs.
+                        CAUTION: only axis == 0 is supported right now.
```
Comment (danpovey) [inline k2/csrc/ragged.h:347]: actually the implementation seems to support axis == 1 right now so this comment isnt right currently

Comment (qindazhu) [reply k2/csrc/ragged.h:347]: Oh yes, we have added `Transpose` in it after the first implementation.

File: k2/csrc/ragged.h (hunk 1 above)
Comment (qindazhu) [reply k2/csrc/ragged.h:342]: Done in PR https://github.com/k2-fsa/k2/pull/250

**Pull request merged**